cpp/include/cudf/io_types.h

```diff
@@ -23,27 +23,36 @@
  */
 #pragma once
 
+ /*
+   * Enumerator for the supported forms of the input CSV file
+   */
+typedef enum 
+{
+  FILE_PATH,								//< Indicates that the input is specified with a file path
+  HOST_BUFFER								//< Indicates that the input is passed as a buffer in host memory
+} gdf_csv_input_form;
```

Comment:
I would've preferred `type` instead of `form` since the latter can give a different meaning, but it's a nitpick.